mysql-test/include/log_encrypt_4.inc

```diff
@@ -70,7 +70,7 @@ DROP TABLE tab1,tab2;
 let NEW_CMD = $MYSQLD $extra_args --initialize-insecure --datadir=$MYSQLD_DATADIR2 --init-file=$BOOTSTRAP_SQL  --secure-file-priv="" --innodb_redo_log_encrypt=$redo_log_mode >>$MYSQLD_LOG 2>&1;
 
 --echo # Run the bootstrap command of datadir2, it should fail since the keyring is not loaded.
---error 1,42
```

Comment:
But 5.7 trunk is also "1,42" here - please check that the null merge does not revert 8.0 to some earlier 5.7 state

Reply:
Double checked with `mtr --debug-server --force log_encrypt_4_rk log_encrypt_4_mk --sanitize`.
42 here is needed to suppress the same `sync_latch_meta_init()` memory leak.

Reply:
Then the state of the current 5.7 trunk and this PR suggests that in 5.7 we reverted sync_latch_meta_init fix (due to upstream, right?), and in 8.0 there is a better fix?

Reply:
No, @laurynas-biveinis . The story was slightly different.
We had `sync_latch_meta_init()` memory leak problem reported in one of our PS-XXX issues.
In https://jira.percona.com/browse/PS-5008 in particular.
I also reported this to upstream (https://bugs.mysql.com/bug.php?id=93165), but they never agreed that this is a real problem and despite all my arguments never agreed to fix this.
Instead in one of the upcoming 5.7 releases they brutally suppressed all these issues by adding `42` to `--error` for a bunch of MTR test cases (I believe the same was done in 8.0 as well).
When we were merging this 5.7 upstream release to Percona Server, we just reverted those suppressions.
Actual memory leak has never been fixed neither in 5.7 nor in 8.0. We still need to do this.

Personally, I am 100% confident that this is the right way to go. MTR code should never include these `--error xxx, 42` ASan suppressions. The main reasons is: once we add this `42` to mark that this test is OK to leak memory, we will never be able to detect if our new changes introduce a new (previously non-existing) leak.

Reply:
OK, then the only remaining issue is that 5.7 HEAD still has --error 42, at least in log_encrypt_4.inc. This also has to be fixed in the scope of this upstream/null merge work.